packages/big-design/src/components/Counter/spec.tsx

```diff
@@ -1,320 +1,335 @@
-import React, { createRef, Ref } from 'react';
+import { screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import React, { createRef, Ref, useState } from 'react';
 import 'jest-styled-components';
 
-import { fireEvent, render } from '@test/utils';
+import { render } from '@test/utils';
 
 import { FormControlDescription, FormControlError, FormControlLabel, FormGroup } from '../Form';
 
-import { Counter, CounterProps } from './index';
+import { Counter } from './index';
 
-interface MockCounterProps extends CounterProps {
-  dataTestId?: string;
+interface MockProps {
   ref?: Ref<HTMLInputElement>;
+  id?: string | undefined;
+  label?: React.ReactChild;
+  labelId?: string;
+  dataTestId?: string;
+  description?: React.ReactChild;
+  value: number;
+  min?: number;
+  step?: number;
+  max?: number;
+  onCountChange?(count: number): void;
+  error?: React.ReactNode | React.ReactNode[];
 }
 
-const val = 5;
-
-const handleChange = jest.fn((num) => num);
-
-const requiredAttributes = {
-  value: val,
-  onCountChange: handleChange,
-};
-
-const counterMock = ({
+const CounterMock: React.FC<MockProps> = ({
   ref,
   label = 'Label',
   labelId = '',
-  id = '',
   error = '',
+  id = '',
   description = '',
-  value,
-  onCountChange,
-  min = 0,
-  max = 10,
+  value = 5,
   dataTestId = '',
-}: MockCounterProps) => (
-  <Counter
-    ref={ref}
-    label={label}
-    labelId={labelId}
-    error={error}
-    id={id}
-    description={description}
-    value={value}
-    data-testid={dataTestId}
-    min={min}
-    max={max}
-    onCountChange={onCountChange}
-  />
-);
+  min = 0,
+  max = 100,
+  onCountChange,
+}) => {
+  const [counterValue, setCounterValue] = useState(value);
+  const handleChange = (value: number) => setCounterValue(value);
+
+  return (
+    <Counter
+      ref={ref}
+      label={label}
+      labelId={labelId}
+      error={error}
+      id={id}
+      description={description}
+      value={counterValue}
+      data-testid={dataTestId}
+      min={min}
+      max={max}
+      onCountChange={onCountChange ?? handleChange}
+    />
+  );
+};
```

Comment:
I wonder if this is better written this way:

```tsx
const CounterMock: React.FC<Partial<ComponentProps<typeof Counter>>> = ({ value = 5, ...props }) => {
  const [counterValue, setCounterValue] = useState(value);
  const handleChange = (value: number) => setCounterValue(value);

  return (
    <Counter
      label="Label"
      error=""
      description=""
      value={counterValue}
      min={0}
      max={100}
      onCountChange={handleChange}
      {...props}
    />
  );
};
```

Reply:
yeah, totally agree.  it looks much cleaner.